src/solvers/flattening/bv_utils.cpp

```diff
@@ -360,6 +360,61 @@ bvt bv_utilst::add_sub(const bvt &op0, const bvt &op1, literalt subtract)
   return result;
 }
 
+bvt bv_utilst::saturating_add_sub(
```

Comment:
Makes sense. Comments could speed up understanding of the algorithm.

Reply:
Fair call, I've added comments in the body of this function.